### ISSUE TITLE
Requesting for separate application data file or database for ApplicationLayerTopology(ALT).

### ISSUE THREAD

**Issue by PrathibaJee** [openBackhaul/ApplicationLayerTopology#117]
**Background :**
In the EATL and OL , two problem we faced , both of them are related to the increased file size , 

1. The initial load file’s size will be < 100 KB
2. On loading more records , the file size will be increased and when it grow and cross 4 MB , the following problem occurs, 
	1. failure in the Frequent CRUD operations from multiple threads
	2. Reading time is directly proportional to the number of records (server starts to respond low when it reach 5 MB of data)

Similarly , for ALT also we face this problem when the file size grows. We had a similar discussion for ALT to have a separate file for Application data. But couldn’t able to see an issue or conversation regarding this.
So creating one.

**Proposal :**
ALT shall store its configuration data in control-construct similar to other applications.
By doing so , we can also get an advantage , 
Currently , for ALT , there is a separate application pattern and basic services package.
This rework shall be avoided if we have a CONFIG file with control-construct information of the ALT application 

ALT shall have the application data that consists of the control-construct of this application as well as the regarded application.
Basically the application data is the current configuration data file which is starting from network-control-domain.
Additionally , we can also better move to a database instead of a file , because of the following reason , 
1. With the increased number(lets say 30 applications) of control-construct instances in the Application data file will make the processing slow.
2. ALT will receive frequent Create, update(especially), delete request from multiple applications. This may lead the same failure we faced for EATL and OL

**Comment by openBackhaul** [comment]
Maybe, ALT will receive less data than EaTL, but data flow towards ALT will be very bursty.
Stabilizing the write operations makes sense. 
The advantage of re-using the BasicServices npm package is also obvious.

**Comment by openBackhaul** [comment]
After making the decision here, ALT issue #132 has been created to document and track the changes that are required for off-loading the topology information into a separate database.
